air/src/execution_step/boxed_value/stream_map.rs

```diff
@@ -0,0 +1,201 @@
+mod test {
+    use super::Generation;
+    use super::StreamMap;
+    use crate::execution_step::boxed_value::stream_map::from_key_value;
+    use crate::execution_step::execution_context::stream_map_key::StreamMapKey;
+    use crate::execution_step::ValueAggregate;
+    use air_trace_handler::merger::ValueSource;
+    use serde_json::json;
+    use std::borrow::Cow;
+    use std::rc::Rc;
+
+    #[test]
+    fn test_from_value() {
+        let obj = json!([{"top_level": [{"first": 42},{"second": 43}]}]);
+        let key_str = "some_key";
+        let key = StreamMapKey::Str(Cow::Borrowed(key_str));
+        let value = Rc::new(obj.clone());
+
+        let generation_idx = 0;
+        let generation = Generation::Nth(generation_idx.into());
+        let value_aggregate: ValueAggregate = ValueAggregate::new(
+            value.clone(),
+            <_>::default(),
+            0.into(),
+            air_interpreter_data::Provenance::literal(),
+        );
+        let stream_map = StreamMap::from_value(key.clone(), &value_aggregate);
+
+        let mut internal_stream_iter = stream_map.stream.iter(generation).unwrap();
+        let v = internal_stream_iter.next().map(|e| e.get_result()).unwrap();
+        let examplar = from_key_value(key, value.as_ref());
+        assert_eq!(*v, examplar);
+        assert_eq!(internal_stream_iter.next(), None);
+
+        let key = StreamMapKey::I64(42.into());
+        let value_aggregate = ValueAggregate::new(
+            value.clone(),
+            <_>::default(),
+            0.into(),
+            air_interpreter_data::Provenance::literal(),
+        );
+        let stream_map = StreamMap::from_value(key.clone(), &value_aggregate);
+
+        let mut internal_stream_iter = stream_map.stream.iter(generation).unwrap();
+        let v = internal_stream_iter.next().map(|e| e.get_result().as_ref()).unwrap();
+        let examplar = from_key_value(key, value.as_ref());
+        assert_eq!(*v, *examplar.as_ref());
+        assert_eq!(internal_stream_iter.next(), None);
+    }
+
+    #[test]
+    fn test_insert() {
+        let obj = json!([{"top_level": [{"first": 42},{"second": 43}]}]);
+        let key_str = "some_key";
+        let key12 = StreamMapKey::Str(Cow::Borrowed(key_str));
+        let value = Rc::new(obj.clone());
+        let generation_idx = 0;
+        let value_aggregate: ValueAggregate = ValueAggregate::new(
+            value.clone(),
+            <_>::default(),
+            0.into(),
+            air_interpreter_data::Provenance::literal(),
+        );
+        let mut stream_map = StreamMap::from_value(key12.clone(), &value_aggregate);
+        let generation = Generation::Nth(generation_idx.into());
+        let generation_idx_res = stream_map
+            .insert(key12.clone(), &value_aggregate, generation, ValueSource::CurrentData)
+            .unwrap();
+        assert_eq!(generation_idx_res, generation_idx);
+
+        let examplar = from_key_value(key12, value.as_ref());
```

Comment:
Fluence general coding standards require using American English in the code.
While `examplar` is considered archaic form of the `exemplar`, I believe the latter should be used.